drivers/include/lm75.h

```diff
@@ -0,0 +1,219 @@
+} lm75_params_t;
+
+/**
+ * @brief lm75 device descriptor
+ */
+typedef struct lm75 {
+    lm75_params_t    lm75_params; /**< Configuration Parameters */
+} lm75_t;
+
+/**
+ * @brief Initialization of the LM75 sensor
+ *
+ * Initializes the sensor according to specific input parameters.
+ *
+ * @param[out] dev        device structure to initialize
+ * @param[in] params      initialization parameters
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ * @return LM75_ERROR, on initialization related error
+ */
+int lm75_init(lm75_t *dev, const lm75_params_t *params);
+
+/**
+ * @brief Temperature values of LM75 sensor
+ *
+ * Reads the sensor temperature values from TEMP_REG
+ * the value is given with the full precision the device is capable of
+ *
+ * @param[in] dev                device structure
+ * @param[in] temperature        buffer where temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_temperature(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Sets the values for Overtemperature shutdown(OS) and Hysteresis temperature(HYST).
+ * OS gives the temperature's higher bound and HYST the lower bound
+ * values are rounded to the lowest value that the device supports,
+ *
+ * @param[in] dev           device structure
+ * @param[in] temp_os       desired OS temperature in mºC
+ * @param[in] temp_hyst     desired HYST temperature in mºC
+ * @param[in] cb            callback that is called from interrupt context
+ * @param[in] *arg          optional arguments for the gpio_init_int function
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ * @return LM75_ERROR, on temperature setting related error
+ */
+int lm75_set_temp_limits(lm75_t *dev, int temp_hyst, int temp_os, gpio_cb_t cb, void *arg);
+
+/**
+ * @brief Overshutdown temperature value of LM75 sensor
+ *
+ * Reads the sensor OS temperature value from TOS_REG in ºC.
+ *
+ * @param[in] dev                 device structure
+ * @param[out] temperature        buffer where OS temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_os_temp(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Hysteresis temperature value of LM75 sensor
+ *
+ * Reads the sensor hysteresis temperature value from THYST_REG in ºC.
+ *
+ * @param[in] dev                 device structure
+ * @param[out] temperature        buffer where HYST temperature value will be written
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR_I2C, on I2C related error
+ */
+int lm75_get_hyst_temp(lm75_t *dev, int *temperature);
+
+/**
+ * @brief Check the current state of the OS pin
+ *
+ * @param[in] dev      device structure
+ *
+ * @return LM75_SUCCESS, on success
+ * @return LM75_ERROR, on pin reading error
+ * @return LM75_I2C_ERROR, on I2C related error
```

Comment:
How can reading a pin cause an I2C error?
What does success mean in this context?

Reply:
That function doesn't just read the pin, but also reads the configuration register to see if the pin is active while low or high. I'll change the description to properly explain this